---
title: "SARS-CoV-2 protein expression"
subtitle: "Final project in R for Bio Data Science (22100)"
author: 
- Alban Laus Obel Slabowska (s200347), Astrid Rasmussen (s193254), Jens Waaben (s205972),
- Signe Katrine Hjort (s174587), Signe Fischer Ravn (s175410)
date: 12-05-2021
output:
  ioslides_presentation:
    widescreen: true
---

```{r setup, include = FALSE}
library(knitr)
library(tidyverse)
knitr::opts_chunk$set(echo = FALSE)
```

# Introduction

## Project background
- SARS-CoV-2 emerged at the end of 2019 as the cause of COVID-19.
- High infectivity has resulted in a still ongoing pandemic.
- There is no current treatment.
- **Currently only little knowledge about the effect of COVID-19 infection on host cell gene expression.**


## Project Aim



*Aim: To carry out tidy data analysis of proteomics data, and communicate the findings to non-bioinformaticians in a way that enables them to get information about their genes of choice.*


# Materials

## Project data

- Data obtained from Bojkova *et al.*
- Infection profile of SARS-CoV-2 using a human cell-culture model.
  - Infection (**Corona**) or **mock**-infection of **Caco-2** (colon epithilial carcinoma cells)


## Project Data

- Protein expression at **four different time** points.
- For each time point we have **triplicates**
- Cells lysed and lysate fed into **LC-MS/MS** to generate **protein counts**. 


## Project Data
- Investigate genes and pathways that are altered during infection.
- Dataset consisting of two files to be joined and cleaned:
  - **Gene counts** and **sample meta-information**.  


# Methods

## Methods

- Everything done in **Tidyverse** R. 
- Done through **Github** to emphasize project collaboration

## Project overview

```{r, echo = FALSE, out.width = "800px"}
include_graphics("/cloud/project/results/flowchart.png")
```

## Handling zeros

- Replace with a **small value**. 
- An inherent problem for -omics and when handling compositional data in general.


## Data augmentation

- After cleaning data was **normalized**
- Normalization is due to different samplesizes
- **FPKM** equivalent **not feasible**
  - **Big genes** are not necessarily translated more 
- On data frame iskept that has **ALL** the replicate information

## Data augmentation

- We calculated **means** for **conditions** over **replicates**. 
- 


- Normalize mean within each group of replicates.
- Mean of replicates
- Log2 fold change between control and infected samples
- Output saved in three files

## heatmap
## top expression
## PCA
## Kmeans







## Analysis

### Log2 fold change
- Log2 fold change
- Top and bottom differentially expressed genes

- Maybe also some focus on why we chose to do so many different scripts for the plots




## Modelling


# Results
## PCA
```{r, echo = FALSE, out.width = "1000px"}
#knitr::include_graphics("/cloud/project/results/04_PCA_plot.png")
```
---

## Heatmap
```{r, echo = FALSE, out.width = "1000px"}
knitr::include_graphics("/cloud/project/results/04_Heatmap.png")
```



## Shiny app
```{r, echo=FALSE}
include_app("https://signe-hjort.shinyapps.io/shiny/", height = "500px")
```


## Project overview and structure
```{r}
include_graphics("/cloud/project/results/flowchart.png")
```


# Discussion
* genes for further
* Benefit if had done gene ontology  ??


